core/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java

```diff
@@ -101,13 +105,31 @@ static void check(
         final List<String> errors = new ArrayList<>();
         final List<String> ignoredErrors = new ArrayList<>();
 
+        final String esEnforceBootstrapChecks = System.getProperty(ES_ENFORCE_BOOTSTRAP_CHECKS);
```

Comment:
commenting here as this is the first line I can comment on. Nit: The java docs of enforceLimits needs to be adapted IMO